app/controllers/placements/providers/placements_controller.rb

```diff
@@ -99,4 +102,20 @@ def compact_school_attribute_values(attribute)
     all_schools.where.not(attribute => nil)
                .distinct(attribute).order(attribute).pluck(attribute)
   end
+
+  def filter_subjects_by_phase
+    if filter_form.primary_selected? && !filter_form.secondary_selected?
```

Comment:
Rather than having `filter_form.primary_selected? && !filter_form.secondary_selected?` and `!filter_form.primary_selected? && filter_form.secondary_selected?` used here and in `filter_schools_by_phase`, consider making methods in the `filter_form`, something like `primary_only?` and `secondary_only?`